### PR TITLE
Add option to specify date in directory structure

### PR DESCRIPTION
**Pull request by Radvendii**
I thought this was an option already. Is there any reason not to allow this as a way of specifying date?

### CODE REVIEW

File: lib/Hakyll/Web/Template/Context.hs [hunk 1]
```diff
@@ -306,6 +312,8 @@ getItemUTC locale id' = do
         [tryField "published" fmt | fmt <- formats] ++
         [tryField "date"      fmt | fmt <- formats] ++
         [parseTime' "%Y-%m-%d" $ intercalate "-" $ take 3 $ splitAll "-" fnCand | fnCand <- reverse paths]
```
Comment (bergus) [inline lib/Hakyll/Web/Template/Context.hs]:
```suggestion
         [parseTime' "%Y-%m-%d" $ intercalate "-" $ take 3 $ splitAll "-" fnCand | fnCand <- reverse paths] ++
```

File: lib/Hakyll/Web/Template/Context.hs [hunk 2]
```diff
@@ -306,6 +312,8 @@ getItemUTC locale id' = do
...
+        [parseTime' "%Y-%m-%d" $ intercalate "-" $ fnCand 
+          | fnCand <- takeWhile ((==3) . length) . map (take 3) . iterate (drop 1) $ paths]
```
Comment (bergus) [inline lib/Hakyll/Web/Template/Context.hs]: Just make use of `Data.List`:
```suggestion
          | fnCand <- map (take 3) $ tails paths]
```
Also didn't you want a `reverse` in here, to prefer the rightmost candidate?

**Comment by Radvendii** [comment]
Thanks @bergus.

Should I squash the commits into one?

**Comment by jaspervdj** [comment]
@Radvendii Don't worry about squashing, I can squash this when merging.  It would be helpful if you could add a test for this though?

**Comment by Radvendii** [comment]
Okay, I finally got this to build on my system, and ran the tests, and it runs without error.

**Comment by Radvendii** [comment]
Does anyone know why it's failing on circleci? The only error I'm seeing is at the end (pasted below), and I'm not sure how that's related to my changes.

```
cabal: The following packages are likely to be broken by the reinstalls:
Cabal-2.4.0.1
libiserv-8.6.3
ghci-8.6.5
ghc-8.6.5
ghc-boot-8.6.5
Use --force-reinstalls if you want to install anyway.

Exited with code 1
```

**Comment by jaspervdj** [comment]
@Radvendii Try merging upstream `master` into your branch

**Comment by Radvendii** [comment]
Okay, it all works now. Anything left to do before merge?

**Comment by jaspervdj** [comment]
Thanks @Radvendii!

**Pull request merged**